### ISSUE TITLE
Loss of the scope on the browser side

### ISSUE THREAD

**Issue by smenshikov** [effector/effector#512]
**What is the current behavior:**
Looks like the scope loses on the browser side

**Please provide the steps to reproduce and if possible a minimal demo of the problem via https://share.effector.dev, https://codesandbox.io or similar**
My app is based on Next.js. Will try describe some components and logic in my app to understand the situation. It is not complex.
So, there are some routes in my app. For now we are interested in two of them: `/account` and `/account/edit`. 
The route `/account/edit` contains component `AccountEdit` like this

<img width="866" alt="Pasted Graphic 1" src="https://user-images.githubusercontent.com/75778692/130226703-4b247e62-06ce-4977-8e5f-cd86d262d36d.png">
So, just a form and effect `updateAccountInfoFx` to send form data to an API. (Just to clarify - the store `$account` fills on the SSR side via an effect that wraps an API endpoint.) Also, there is the `sample`

<img width="468" alt="sample( config" src="https://user-images.githubusercontent.com/75778692/130226761-972c80ed-6a05-4437-8c0d-273f6147a162.png">
So, after successfull sending to API will be displayed a notification in UI. `setNotification` fills the store `$notification`. 

There is a component `DefaultLayout` that wraps all pages and render the notification like this

<img width="932" alt="Pasted Graphic 4" src="https://user-images.githubusercontent.com/75778692/130226788-80f4d87b-e554-4fcf-9828-d658769d701d.png">
 The introductory information is over. Now will be the issue. 
If we open the route `/account/edit` directly and update data in the form then the notification will be displayed. Here is a short demo 
[iCloud link 1](https://www.icloud.com/iclouddrive/04djR8FExAUpheAckrYJx0a4g#%D0%97%D0%B0%D0%BF%D0%B8%D1%81%D1%8C_%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0_2021-08-20_%D0%B2_13.49.08)

If we open the route `/account` and then go to the route `/account/edit` via Next.js routing, then again update data in the form, so after submitting the notification will not be displayed. Here is a short demo 
[iCloud link 2](https://www.icloud.com/iclouddrive/0gC4YRPTBB0ahFM_tROOSmx4A#%D0%97%D0%B0%D0%BF%D0%B8%D1%81%D1%8C_%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0_2021-08-20_%D0%B2_13.50.56)

Hope everything was clear. 

**What is the expected behavior:**
The scope should be the same on the browser side and SSR side

**Which versions of effector packages, and which browser and OS are affected by this issue? Did this work in previous versions of effector?:**
The issue is available in effector version 21.8.12-b75df7cb and effector-react 21.3.3-b75df7cb. 
There is no issue in the version effector 21.8.11 and effector-react 21.3.2

**Comment by zerobias** [comment]
Did you use `serialize(scope)` _without_ `onlyChanges: true`? Because it’s looks like relying on server side data for resetting client side state. 

I could be more specific if there was some reproduce I could run. From the description alone it is difficult to understand what exactly was used in application

**Comment by smenshikov** [comment]
@zerobias I use `serialize` function with `onlyChanges: true` in the `_app.tsx` file
![image](https://user-images.githubusercontent.com/75778692/130245463-a7f2c725-229f-413f-bac5-062d4ac0022f.png)

**Comment by zerobias** [comment]
is there onlyChanges on server side? it was critical to not to send unrelated defaults:

let’s assume there are pages `/dashboard` and `/profile`, if your server side code was working without that flag, it will lead to obscure behavior:

client initialized on route `/profile`, it’s serialized state will looks like that:

```json
{
  "profile": {"user": "alice", "offset": 0},
  "dashboard": null
}
```

then user do something on `/profile` page:

```json
{
  "profile": {"user": "alice", "offset": 10},
  "dashboard": null
}
```

then client go to `/dashboard`, and that’s how server will behave without onlyChanges:

_I’m a dashboard route, I have nothing in any unrelated store, so they have default values. I’m not going to send only my changes, so I’ll send defaults too_:

```json
{
  "profile": null,
  "dashboard": {"title": "welcome"}
}
```

which will __erase__ all unrelated stores on client

this behavior wasn’t only insidious, but also so inconvenient, that it was impossible to implement in new fork, so canary builds and effector 22 will always work in onlyChanges: true mode

I suggest to explicitly manage notification state: call `hideNotification` event during client-side navigation or something like that to be sure that notification store would not stuck

Real loss of scope can be differentiated by `.watch(console.log)`: loss of scope keep watch being called, in case of lack of onlyChanges nothing will be called at all

**Comment by zerobias** [comment]
If you still think this is a loss of scope then reproduce is really important because then I entirely unaware about that case

**Comment by smenshikov** [comment]
@zerobias Will try make a repo with the issue

**Comment by smenshikov** [comment]
@zerobias I can not reproduce the issue. It works fine,  but could you clarify how I can detect that scope is lost because in my project still has strange behavior

**Comment by zerobias** [comment]
run this code in both canary build and npm version

```ts
function MyApp(…) {
  // in the beginning
  console.log('$notification', $notification.sid)
  if (clientScope)
    console.log('current scope', serialize(clientScope, {onlyChanges: true}))
  console.log({initialState, store})
  
  const scope = …

  console.log('new scope', serialize(scope, {onlyChanges: true}))
}

updateAccountInfoFx.finally.watch(status => {
  console.log('updateAccountInfo', status)
})

setNotification.watch(upd => {
  console.log('setNotification', upd)
})
$notification.updates.watch(upd => {
  console.log('notification changed', upd)
})
```

also need to print what your getServerSideProps/getStaticProps has after serialize call

one more thing: check that whole application use __only__ `@effector/effector` and `@effector/effector-react`, especially in babel-config

**Comment by zerobias** [comment]
In case of complex bugs reproduction does not mean "recreation", but rather "reduction": didn’t imitate error from scratch, take your real project and drop everything unrelated, usually it’s easier than dealing with blank project; 90% of code will disappear right after removing unrelated `/pages`

**Comment by smenshikov** [comment]
@zerobias I use canary versions in the package.json file this way
![image](https://user-images.githubusercontent.com/75778692/130283664-d5060fff-6f01-489f-a9d3-0b64935ef197.png)
So, I have downloaded them and added the local path. Both packages are in the root directory of the project. The .babelrc file looks like this 
![image](https://user-images.githubusercontent.com/75778692/130283916-849fa0c9-ecbc-459e-9bfd-c0bb88c2da03.png)

So, the whole app doesn't have a difference for import declarations from effector. The import declarations are `import {...} from 'effector'` or `import {...} from 'effector-react/ssr'` as for npm versions

**Comment by zerobias** [comment]
Okay 👌 Then we need to compare logs from original effector and current canary build as described [above](https://github.com/effector/effector/issues/512#issuecomment-902871397)

**Comment by smenshikov** [comment]
@zerobias Finally I created the reproduce repo https://github.com/smenshikov/effector-issue. Steps to reproduce: 

1. open http://localhost:3000/client
2. click to link **client2 page**
3. click on the button **Fill $notification store**

**Comment by smenshikov** [comment]
@zerobias If you open the link http://localhost:3000/client2 and then click on the button **Fill $notification store** in this case it works fine

**Comment by smenshikov** [comment]
@zerobias As for me problem with `Layout` component. If you move `Layout` from `_app` file to each page then it works fine

**Comment by zerobias** [comment]
Fixed in canary build [a7ac3b66](https://github.com/effector/effector/packages/21591?version=21.8.12-a7ac3b66)
